model.py

```diff
@@ -9,43 +9,98 @@
 import utils
 
 
+class GaussianAttention(nn.Module):
+    def __init__(self, query_size=5, use_position=True):
+        super(GaussianAttention, self).__init__()
+        self.linear = nn.Linear(in_features=query_size, out_features=2)
+        self.use_position=use_position
+
+    def forward(self, query, textEmbedding, text_mask, position):
+        """query should be of shape [seq_len, batch_size, query_size]]
+        textEmbedding should be of shape [text_len, batch_size]
+        text_mask should be of shape [text_len, batch_size]
+        position should be of shape [seq_len, batch_size] (or None)
+        Returns tuple of:
+          * a tensor of shape [seq_len, batch_size, embedding_dim]
+          * a tensor of shape [seq_len, batch_size] containing positions."""
+        # TODO(neil): rewrite so that we don't need seq_len leading dimension
+        maskedEmbedding = textEmbedding * text_mask.unsqueeze(-1)
+        seq_len = query.shape[0]
+        batch_size = query.shape[1]
+        if position is None:
+            position = torch.zeros((seq_len, batch_size), device=query.device)
+        query_result = torch.exp(self.linear(query))
+        assert query_result.shape == (seq_len, batch_size, 2)
+        assert textEmbedding.shape[1] == batch_size
+        mean, std = query_result.unbind(dim=-1)
+        if self.use_position:
+            mean = position + mean.cumsum(dim=0)/20
+        weights = torch.zeros((seq_len, textEmbedding.shape[0], batch_size, 1), device=query.device)
```

Comment:
I think that this line isn't necessary anymore.

Reply:
Removed